react/features/analytics/handlers/amplitude/fixDeviceID.native.ts

```diff
@@ -14,11 +17,17 @@ export async function fixDeviceID(amplitude: Amplitude) {
     const current = await DefaultPreference.get('amplitudeDeviceId');
 
     if (current) {
-        amplitude.setDeviceId(current);
+        await amplitude.setDeviceId(current);
     } else {
-        const uid = DeviceInfo.getUniqueId();
+        syncUniqueId().then(uid => {
```

Comment:
syncUniqueId does not return a promise!

Reply:
https://github.com/react-native-device-info/react-native-device-info#syncuniqueid
From their doc, it does.

Reply:
Ah, their API is kinda confusing there. This is the one we want to be using: https://github.com/search?q=repo%3Areact-native-device-info%2Freact-native-device-info%20getUniqueIdSync&type=code

Reply:
Or just getUniqueId and await it.